File: leetsync/2478-longest-nice-subarray/longest-nice-subarray.go

func longestNiceSubarray(nums []int) int {
	n := len(nums)
	left, right, result, mask := 0, 0, 1, 0
	for right < n {
		for (mask & nums[right]) != 0 {
			mask = (mask ^ nums[left])
			left++
		}
		result = max(result, right-left+1)
		mask = (mask | nums[right])
		right++
	}
	return result
}

File: leetsync/2478-longest-nice-subarray/longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int n = nums.length;
        int left = 0, right = 0, result = 1, mask = 0;
        while (right < n) {
            while ((mask & nums[right]) != 0) {
                mask = (mask ^ nums[left]);
                left++;
            }
            result = Math.max(result, right - left + 1);
            mask = (mask | nums[right]);
            right++;
        }
        return result;
    }
}